quarkus.jdt.ext/com.redhat.microprofile.jdt.quarkus/src/main/java/com/redhat/microprofile/jdt/internal/quarkus/scheduler/SchedulerUtils.java

```diff
@@ -0,0 +1,127 @@
+/*******************************************************************************
+* Copyright (c) 2021 Red Hat Inc. and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+* which is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.microprofile.jdt.internal.quarkus.scheduler;
+
+import java.time.Duration;
+import java.time.format.DateTimeParseException;
+import java.util.regex.Pattern;
+
+/**
+ * Utility class for quarkus scheduler
+ *
+ */
+public class SchedulerUtils {
+
+	// Patterns for each cron part of the cron member expression in order of
+	// SchedulerErrorCode enum ordinal
+	// Note: DAY_OF_MONTH and DAY_OF_WEEK are mutually exclusive
+
+	private static final String DAY_OF_MONTH_ABBREV = "^MON$|^TUE$|^WED$|^THU$|^FRI$|^SAT$|^SUN$";
+	private static final String MONTH_ABBREV = "^JAN$|^FEB$|^MA[RY]$|^APR$|^JU[NL]$|^AUG$|^SEP$|^OCT$|^NOV$|^DEC$";
+	private static final Pattern[] CRON_PATTERNS = {
+			Pattern.compile("(^[1-5]?[0-9]$|\\*|\\?|^\\*\\/\\d+)"), // SECONDS
+			Pattern.compile("(^[1-5]?[0-9]$|\\*|\\?|^\\*\\/\\d+)"), // MINUTES
+			Pattern.compile("(^(2[0-3]|1[0-9]|[1-9])$|\\*|\\?|^\\*\\/\\d+)"), // HOUR
```

Comment:
This doesn't support 0, although the diagnostic text does mention 0 is a valid value.